include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -483,46 +456,50 @@ namespace internal
               zero_tol_hessian)
             return false;
 
-      const unsigned int stride = (n_q_points_1d + 1) / 2;
-      shape_values_eo.resize((fe_degree + 1) * stride);
-      shape_gradients_eo.resize((fe_degree + 1) * stride);
-      shape_hessians_eo.resize((fe_degree + 1) * stride);
-      for (unsigned int i = 0; i < (fe_degree + 1) / 2; ++i)
-        for (unsigned int q = 0; q < stride; ++q)
-          {
-            shape_values_eo[i * stride + q] =
-              0.5 * (shape_values[i * n_q_points_1d + q] +
-                     shape_values[i * n_q_points_1d + n_q_points_1d - 1 - q]);
-            shape_values_eo[(fe_degree - i) * stride + q] =
-              0.5 * (shape_values[i * n_q_points_1d + q] -
-                     shape_values[i * n_q_points_1d + n_q_points_1d - 1 - q]);
-
-            shape_gradients_eo[i * stride + q] =
-              0.5 *
-              (shape_gradients[i * n_q_points_1d + q] +
-               shape_gradients[i * n_q_points_1d + n_q_points_1d - 1 - q]);
-            shape_gradients_eo[(fe_degree - i) * stride + q] =
-              0.5 *
-              (shape_gradients[i * n_q_points_1d + q] -
-               shape_gradients[i * n_q_points_1d + n_q_points_1d - 1 - q]);
-
-            shape_hessians_eo[i * stride + q] =
-              0.5 * (shape_hessians[i * n_q_points_1d + q] +
-                     shape_hessians[i * n_q_points_1d + n_q_points_1d - 1 - q]);
-            shape_hessians_eo[(fe_degree - i) * stride + q] =
-              0.5 * (shape_hessians[i * n_q_points_1d + q] -
-                     shape_hessians[i * n_q_points_1d + n_q_points_1d - 1 - q]);
-          }
-      if (fe_degree % 2 == 0)
-        for (unsigned int q = 0; q < stride; ++q)
-          {
-            shape_values_eo[fe_degree / 2 * stride + q] =
-              shape_values[(fe_degree / 2) * n_q_points_1d + q];
-            shape_gradients_eo[fe_degree / 2 * stride + q] =
-              shape_gradients[(fe_degree / 2) * n_q_points_1d + q];
-            shape_hessians_eo[fe_degree / 2 * stride + q] =
-              shape_hessians[(fe_degree / 2) * n_q_points_1d + q];
-          }
+      auto convert_to_eo = [](AlignedVector<Number> &      array_eo,
+                              const AlignedVector<Number> &array,
+                              const unsigned               n_points_dst,
+                              const unsigned               n_points_src) {
+        const unsigned int stride = (n_points_dst + 1) / 2;
```

Comment:
You will need to rule out the case where `array.empty()` which is the case of `n_q_points_1d >= 200` at this stage. While here, I realized that we might want to change the lambda so it returns the `array_eo` instead? This will allow you to write `shape_values_eo = convert_to_eo(shape_values, n_q_points_1d, fe_degree+1)`.